plugins/auto-sizes/includes/improve-calculate-sizes.php

```diff
@@ -187,9 +189,14 @@ function auto_sizes_calculate_better_sizes( int $id, string $size, string $align
 
 		case 'left':
 		case 'right':
-			// These alignment get constrained by the wide layout size but do not get stretched.
-			$alignment    = auto_sizes_get_layout_width( 'wide' );
-			$layout_width = sprintf( '%1$spx', min( (int) $alignment, $image_width ) );
+			/*
+			 * Update width for cover block.
+			 * See https://github.com/WordPress/gutenberg/blob/938720602082dc50a1746bd2e33faa3d3a6096d4/packages/block-library/src/cover/style.scss#L82-L87.
+			 */
+			if ( 'core/cover' === $block_name ) {
+				$image_width = $image_width * 0.5;
```

Comment:
Good idea! I think that instead we could pass this as the function as the `$size` or `$resize_width` value for `auto_sizes_calculate_better_sizes()`  and avoid needing to add another argument to the function signature. Alternately, we could consider allowing arbitrary values to be passed to the `$max_alignment` value, which then overrides any alignment value, which may come in handy whenever we want to pass a maximum width value based on things like columns or grid values.

Also, small nitpick, but this calculation isn't the same as what is used for the cover block, instead it calculates 0.5 * `$content-width`, which is a CSS variable that is defined [in this package](https://github.com/WordPress/gutenberg/blob/938720602082dc50a1746bd2e33faa3d3a6096d4/packages/base-styles/_variables.scss#L125). So it should always end up being '420px'.

Reply:
Went with the $size idea and took advantage of the fact that you can use an array for $size when passing to `wp_get_attachment_image_src()`. See, [d3a60d0](https://github.com/WordPress/performance/pull/1701/commits/d3a60d0ab386fde7724f4f7e7de0fa6f102c6b6b).